### PR TITLE
TST: Test Loc to set Multiple Items to multiple new columns

### LINKED ISSUE #25594: Adding multiple columns with loc fails
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

df = pd.DataFrame(index=[1, 2], columns = list('ab'))

# Single assignment on existing column
df.loc[1, 'a'] = 1

# Multiple assignment on existing column
df.loc[2, ['a', 'b']] = [1, 2]

# Single assignment on new column
df.loc[3, 'c'] = 3

# Multiple assignment on single new column
df.loc[[1, 2], 'l'] = [6, 7]

# Multiple assignment on multiple new column
# Surprisingly fails
try:
    df.loc[3, ['d', 'e']] = [6, 7]
    print(df)
except KeyError:
    print("Error")

```
#### Problem description
The behaviour of loc is surprising:
- When assigning, new columns and index are created as required.
- You can assign multiple index and columns
- BUT, if you try to create a column or index while doing multiple assignment, new columns and index are NOT created as required, and you get a KeyError.

Futhermore, using the indexing operators `[]` directly will create the required columns.

#### Expected Output
```
     a    b    c    l    d    e
1    1  NaN  NaN  6.0  NaN  NaN
2    1    2  NaN  7.0  NaN  NaN
3  NaN  NaN  3.0  NaN  6.0  7.0
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.2.final.0
python-bits: 64
OS: Darwin
OS-release: 18.2.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.1
pytest: 4.2.0
pip: 19.0.2
setuptools: 40.8.0
Cython: 0.28.5
numpy: 1.15.2
scipy: 1.2.0
pyarrow: None
xarray: None
IPython: 7.0.1
sphinx: 1.8.1
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: 4.2.5
bs4: None
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by GYvan**
- [x] closes #25594
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry
added test to pandas/tests/indexing/multi/test_loc.py

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @GYvan for the PR. There are some lint failures and another git commit.

see https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#working-with-the-code

**Comment by GYvan** [comment]
> Thanks @GYvan for the PR. There are some lint failures and another git commit.
> 
> see https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#working-with-the-code

Thanks, I will look through them

**Comment by GYvan** [comment]
> Thanks @GYvan for the PR. There are some lint failures and another git commit.
> 
> see https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#working-with-the-code

**Comment by GYvan** [comment]
> > Thanks @GYvan for the PR. There are some lint failures and another git commit.
> > see https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#working-with-the-code



> Thanks @GYvan for the PR. There are some lint failures and another git commit.
> 
> see https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#working-with-the-code

**Comment by GYvan** [comment]
Hi @simonjayhawkins. I have tried to address the lint failures and reverted the unrelate changes. Can you please let me know if anything else needs changes?

**Comment by jbrockmendel** [comment]
cc @phofl

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -2768,3 +2768,15 @@ def test_loc_setitem_dict_timedelta_multiple_set(self):
...
+        # GH 25594
+        df = DataFrame(index=[1, 2], columns=["a"])
+        df.loc[1, ["b", "c"]] = [6, 7]
+        result = df.copy()
+
+        expected = DataFrame(index=[1, 2], columns=["a", "b", "c"])
+        expected.loc[1, "b"] = 6
+        expected.loc[1, "c"] = 7
+
+        tm.assert_frame_equal(result, expected, check_dtype=False)
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py:2782]: does this fail w/o the check_dtype?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexing/test_loc.py:2782]: Yes, as it should. The expected should have dtype object while df has dtype float in these columns

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -2768,3 +2768,15 @@ def test_loc_setitem_dict_timedelta_multiple_set(self):
             [[Timedelta(6, unit="s"), "foo"]], columns=["time", "value"], index=[1]
         )
         tm.assert_frame_equal(result, expected)
+
+    def test_loc_set_multiple_items_in_multiple_new_columns(self):
+        # GH 25594
+        df = DataFrame(index=[1, 2], columns=["a"])
+        df.loc[1, ["b", "c"]] = [6, 7]
+        result = df.copy()
```
Comment (phofl) [inline pandas/tests/indexing/test_loc.py:2776]: No result needed, just use df

File: pandas/tests/indexing/test_loc.py [hunk 3]
```diff
@@ -2768,3 +2768,15 @@ def test_loc_setitem_dict_timedelta_multiple_set(self):
...
+
+        expected = DataFrame(index=[1, 2], columns=["a", "b", "c"])
```
Comment (phofl) [inline pandas/tests/indexing/test_loc.py:2778]: Please define expected without the use of loc. You can show the result of the sequential operations too, but please use another check for this

**Review by phofl — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py [hunk 4]
```diff
@@ -2773,10 +2773,7 @@ def test_loc_set_multiple_items_in_multiple_new_columns(self):
...
         df.loc[1, ["b", "c"]] = [6, 7]
-        result = df.copy()
 
-        expected = DataFrame(index=[1, 2], columns=["a", "b", "c"])
-        expected.loc[1, "b"] = 6
-        expected.loc[1, "c"] = 7
+        expected = DataFrame({"a": [np.nan, np.nan], "b": [6, np.nan], "c": [7, np.nan]}, index = [1, 2])
 
-        tm.assert_frame_equal(result, expected, check_dtype=False)
+        tm.assert_frame_equal(df, expected, check_dtype=False)
```
Comment (phofl) [inline pandas/tests/indexing/test_loc.py:2781]: Please remove the check_dtype=False parameter

Comment (phofl) [reply pandas/tests/indexing/test_loc.py:2781]: Then you have to define expected accordingly

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @phofl merge when ready

**Comment by phofl** [comment]
Thanks @GYvan

**Pull request merged**